doc/maintenance.md

```diff
@@ -0,0 +1,538 @@
+- MUST include in the request body, one or more Maintainable Artefacts in the SDMX format indicated in the `Content-type` header and of the SDMX type indicated in the resource, i.e.:
+  - any set of Maintainable Artefacts under resource `/structure/`
+  - a set of specific type of Maintainable Artefacts under the corresponding resource type, e.g. for Codelists: `/structure/codelist/`
+
+##### Server
+
+In response to Artefact(s) creation, the server:
+
+- MUST return `201` upon successful creation (or `207` for partial success);
+- MUST return a `SubmitStructureResponse` message with the result of the action(s), according to the `Accept` header, or the default, if the `Accept` type is not supported (currently available only in SDMX-ML format);
+- MUST set the `Content-type` according to the returned format;
+- MAY set the `Location` header to point to the created/primary resource/Artefact (only one instance is allowed);
+
+##### Response
+
+The `SubmitStructureResponse` message must be returned in any case (success, partial success, failure).
+This is defined as part of the `RegistryInterface` messages.
+The details of the message are explained in section **Structure response message**, below.
+
+Especially when different results occur on the Artefacts (e.g. partial success), the server should act as follows:
+- Return a multi-status return code (like `207`);
+- Return a `JSON` or `XML` message with the details of the result (currently available only in SDMX-ML);
+
+| Method | Exists | Versioning respected | Is Referenced | References exist/provided | Response Code |
+|---|---|---|---|---|---|
+| POST | F | T | - | T | `201` (successful) or `207` (partially successful) |
+| POST | F | T | - | F | `409` missing references |
+| POST | F | F | - | I | `409` failure to respect the versioning rules |
+
+<sup>T: True, F: False, I: Irrelevant, -: Not applicable</sup>
+
+
+### UPDATE
+
+Updating existing SDMX Structural Metadata is detailed in this section.
+
+#### Replace full Artefacts
+
+Following the current SDMX practices, updating (replacing) means providing the new version of any Maintainable Artefact.
+According to RFC7231, `PUT` is the proper way to update a resource, as identified by the URL, but `POST` may also be used in case more than one resources need to be updated.
+
+The `POST` method may be used like this:
+
+- under `/structure` for different types of Maintainable Artefacts, i.e. an SDMX Structure message; 
+- under `/structure/{maintainable}` for Maintainable Artefacts of type `{maintainable}`, i.e. an SDMX Structure message including only a specific type of Structures; e.g. an SDMX message with one or more Dataflows, under `/structure/dataflow`;
+
+The `PUT` method may be used like this:
+
+- under `/structure/{maintainable}/{identifier}` for the Maintainable Artefact of type `{maintainable}`, identified by `{identifier}`, i.e. an SDMX DSD message with `ESTAT:NA_MAIN(1.1)` under `/structure/datastructure/ESTAT/NA_MAIN/1.1`
```

Comment:
Just checking the syntax, you have {maint}/{ident} and no version, but you have a version in the example - is this correct?  I'm not sure what an SDMX DSD message is - is this an SDMX Structure message containing a single DSD, or can they submit a message with the DSD on its own?

Reply:
{maint} is supposed to be the artefact type, eg `datastructure` and {identifier} is supposed to be a full identifier (agencyId/id/version) of an Artefact. Maybe a better naming could help.

Reply:
For PUT, normally you must use exactly the same resource URL as you would expect to use for the subsequent GET. Thus an artefact OECD:DF(1.0) must be put under /OECD/DF/1.0, and a non-versioned artefact OECD:DF() must be put under /OECD/DF/.
By the way, frequently the {identifier} is often understood as the ID of the artefact, e.g. "NA_MAIN". I have seen full identifiers being abbreviated by {ref}.